src/swap.ts

```diff
@@ -0,0 +1,329 @@
+import { purry } from './purry';
+import { isString } from './isString';
+import { isArray } from './isArray';
+import { clone } from './clone';
+
+type ParseInt<T> = T extends `${infer N extends number}` ? N : never;
+
+type Length<T extends ReadonlyArray<any>> = T extends { length: infer L }
+  ? L
+  : never;
```

Comment:
I'm not very skilled with TypeScript type operations, so I largely used code that I copied and pasted from a blog. Because it was code from the [blog](https://itnext.io/implementing-arithmetic-within-typescripts-type-system-a1ef140a6f6f), I didn't thoroughly check this part. I'm sorry for any confusion. I promise to pay more attention and verify everything more closely in the future.

Reply:
The problem with using a blog (and AI in fact) is that the code represents what's the right way to do things in that specific period of time. The speed at which typescript moves, a blog post from 2020 just isn't current enough, and because it's not actual code that runs, just blobs of text, nothings there to make sure it stays up-to-date or even continues working.